src/modules/endpoint/DefaultDataContainer.js

```diff
@@ -24,71 +25,74 @@ class DefaultDataContainer implements DataContainer {
     this._databaseConnector = new DatabaseConnector()
   }
 
+  removeOutdated (newContext: DatabaseContext) {
+    if (this._currentContext && !newContext.equals(this._currentContext)) {
+      this._lastUpdate = null
+      this._categoriesMap = null
+      this._resourceCache = null
+      this._events = null
+    }
+
+    this._currentContext = newContext
+  }
+
   getCities = async (): Promise<Array<CityModel>> => {
     if (this._cities === null) {
       throw Error('Cities are null.')
     }
     return this._cities
   }
 
-  getCategoriesMap = async (): Promise<CategoriesMapModel> => {
-    if (this._context === null) {
-      throw Error('Context has not been set yet.')
-    }
+  getCategoriesMap = async (context: DatabaseContext): Promise<CategoriesMapModel> => {
+    this.removeOutdated(context)
+
     if (this._categoriesMap === null) {
-      throw Error('CategoriesMap is null.')
+      this._cities = await this._databaseConnector.loadCategories(context)
     }
     return this._categoriesMap
   }
 
-  getEvents = async (): Promise<Array<EventModel>> => {
-    if (this._context === null) {
-      throw Error('Context has not been set yet.')
-    }
+  getEvents = async (context: DatabaseContext): Promise<Array<EventModel>> => {
+    this.removeOutdated(context)
+
     if (this._events === null) {
-      throw Error('Events are null.')
+      this._events = await this._databaseConnector.loadCategories(context)
```

Comment:
I guess loadEvents() should be called here

Reply:
good catch! you are right